### ISSUE TITLE
Flags specifics to a language mode are not diagnosed in other language modes

### ISSUE THREAD

**Issue by cor3ntin** [llvm/llvm-project#139058]
```
clang -x c -fexceptions -frtti -fchar8_t -fcoroutines -fcxx-modules
```

We do not emit diagnostics for `-fexceptions -frtti -fchar8_t`, which are not valid C options.

(but we seem to do for `-fcoroutines -fcxx-modules`)

https://compiler-explorer.com/z/48hc6Krh3
